dashboard/app/models/concerns/serialized_properties.rb

```diff
@@ -10,7 +10,7 @@ module SerializedProperties
     self.serialized_properties ||= {}
 
     after_initialize :init_properties
-    before_save {properties.select! {|_, v| v.present?}}
+    before_save {properties.select! {|_, v| !v.nil?}}
```

Comment:
Does this line benefit us beyond the dark mode change? It looks like this was in the original site code, so there isn't much documentation, but before making this change, we should understand why this line may have been added originally and what code depends on it. Reading up on [present vs nil](https://blog.appsignal.com/2018/09/11/differences-between-nil-empty-blank-and-present.html), nil will only catch non-existent values whereas present is far more permissive.

Reply:
Another possibility: We're likely to add more view modes. i.e. in addition to dark and light mode, we will very likely also add high contrast down the line. Would it make sense to make dark vs light an enum rather than a bool? In that case, a value would always be stored and we will be a step closer to being able to add more modes.

Reply:
Yeah, this fixes the issue for dark mode but could potentially have some other consequences especially given how old it is. I like the idea of changing this to an enum, which aligns with Molly's suggestion as well. I think that's probably the safer thing to do if we're considering more view modes down the line.